oxidauth-http/src/server/api/v1/auth/oauth2/redirect.rs

```diff
@@ -0,0 +1,45 @@
+use axum::{Json, extract::State, response::IntoResponse};
+use serde::{Deserialize, Serialize};
+use tracing::info;
+
+use oxidauth_kernel::{auth::oauth2::redirect::*, error::IntoOxidAuthError};
+
+use crate::{provider::Provider, response::Response};
+
+pub type Oauth2RedirectReq = Oauth2RedirectParams;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Oauth2RedirectRes {
+    pub redirect_url: String,
+}
+
+#[tracing::instrument(name = "oauth2_redirect_handler", skip(provider))]
+pub async fn handle(
+    State(provider): State<Provider>,
+    Json(params): Json<Oauth2RedirectParams>,
+) -> impl IntoResponse {
+    let service = provider.fetch::<Oauth2RedirectService>();
+
+    let result = service.call(&params).await;
+
+    match result {
+        Ok(res) => {
+            info!(
+                message = "successfully return oauth2 redirect url",
+                response = ?res,
+            );
+
+            Response::success().payload(Oauth2RedirectRes {
+                redirect_url: res.redirect_url.to_string(),
```

Comment:
Why `to_string`?

Reply:
Doesn't have to be. This has been updated to type Url - no to_string required